### PR TITLE
 Add yi model example

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Add yi model example. Since Yi model is not supported by transformers, use `gpt2` model type for ort optimization pass for now.

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: examples/yi/config.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    "passes": {
+        "conversion": {
+            "type": "OnnxConversion",
+            "config": {
+                "target_opset": 13,
+                "save_as_external_data": true,
+                "all_tensors_to_one_file": true
+            }
+        },
+        "optimize": {
```
Comment (trajepl) [inline examples/yi/config.json:40]: also enable external saving flags?

File: examples/yi/config.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            "config": {
+                "model_type": "gpt2",
+                "float16": true
+            }
+        }
+    },
+    "engine": {
+        "search_strategy": false,
+        "evaluate_input_model": false,
+        "host": "local_system",
```
Comment (trajepl) [inline examples/yi/config.json:52]: nit: local system can be ignored.

File: examples/yi/user_script.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import torch
+
+
+def dummy_inputs(model):
+    attention_mask_sequence_length = 19
+    sequence_length = 19
+    return {
+        "input_ids": torch.randint(10, (1, sequence_length), dtype=torch.int64),
+        "attention_mask": torch.randint(10, (1, attention_mask_sequence_length), dtype=torch.int64),
+    }
```
Comment (trajepl) [inline examples/yi/user_script.py:10]: Any reason we skip the kv cache inputs?

**Comment by xiaoyu-work** [comment]
Closing.

**Pull request closed without merging**